server/src/main/java/org/solarbank/server/dto/EnergyTariff.java

```diff
@@ -1,17 +1,28 @@
 package org.solarbank.server.dto;
 
+import com.fasterxml.jackson.databind.PropertyNamingStrategies;
+import com.fasterxml.jackson.databind.annotation.JsonNaming;
+import javax.money.CurrencyUnit;
+import javax.money.Monetary;
+import javax.money.UnknownCurrencyException;
 import jakarta.validation.constraints.DecimalMin;
-import jakarta.validation.constraints.NotBlank;
 import jakarta.validation.constraints.NotNull;
-import jakarta.validation.constraints.Size;
 import lombok.Data;
 
 @Data
+@JsonNaming(PropertyNamingStrategies.UpperCamelCaseStrategy.class)
 public class EnergyTariff {
-    @NotBlank
-    private String currencyCode;
+    private CurrencyUnit currencyCode;
 
-    @NotNull
+    @NotNull(message = "energy tariff must be provided")
     @DecimalMin(value = "0.01", message = "amount must be at least 0.01")
     private double amount;
+
+    public CurrencyUnit getCurrencyCode() {
```

Comment:
used CurrencyUnit here to validate the currency code, it throws an UnknownCurrencyUnit exception if not a recognised currency, however the exception gets picked up by the DefaultExceptionHandler rather than the UnkownCurrencyHandler

Reply:
Went for option 1, added custom validator